### ISSUE TITLE
SparseCrossEntropy in TensorFlow

### ISSUE THREAD

**Issue by LeavesBreathe** [tensorflow/tensorflow#420]
Hey TF, 

In your Translate, seq2seq example, there is a part where you require conversion from a sparse to dense matrix to do a softmax. 

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/seq2seq.py#L670

The comment there states that tensorflow needs a sparsecrossentropy function in order for this computation to be done on the gpu (and more efficiently overall). I was wondering: was this implementation planned to happen soon? Thanks!

**Comment by lukaszkaiser** [comment]
It's on our list of things to do, yes, the sooner the better. Meanwhile the one in seq2seq seems to be reasonably efficient, at least it's not a bottleneck - check it out.

**Comment by LeavesBreathe** [comment]
thanks @lukaszkaiser -- I have been using the one in seq2seq since its release. The biggest problem though is that you are forced to used a cpu for the softmax, meaning you can't take advantage of using another gpu to do the softmax. As a result, it does lead to longer training times. 

I'll close this issue. But I wouldn't mind trying to help out to get this done on the gpu! Thanks!
